src/tests/ftest/security/pool_connect_init.py

```diff
@@ -0,0 +1,158 @@
+
+    def poolcreate(self, user_uid, user_gid):
+        """
+        pool create method, use pool mode, setid, and size from yaml.
+        Args:
+            user_uid: user or owner user id.
+            user_gid: user or owner group id.
+        Return:
+            pool handle.
+        """
+        # get parameters from yaml file
+        mode = self.params.get("mode", '/run/poolparams/createmode/', 511)
+        setid = self.params.get("setname", '/run/poolparams/createset/',
+                                "daos_server")
+        size = self.params.get("size", '/run/poolparams/createsize/',
+                               107374182)
+        self.log.info("===Create DAOS pool with:")
+        self.log.info("===uid:  %s", user_uid)
+        self.log.info("===gid:  %s", user_gid)
+        try:
+            self.pool = DaosPool(self.context)
+            self.pool.create(mode, user_uid, user_gid, size, setid)
+            self.log.info("===>Pool create succeed.")
+        except DaosApiError as exc:
+            self.log.info(exc)
+            self.log.info(traceback.format_exc())
+            self.fail("##(0)Pool create failed in setUp.\n")
+        return self.pool
+
+    def tearDown(self):
+        """
+        tear down method
+        """
+        try:
+            if self.pool1 is not None:
+                self.pool1.disconnect()
+                self.pool1.destroy(1)
+            if self.pool2 is not None:
+                self.pool2.disconnect()
+                self.pool2.destroy(1)
+            if self.pool3 is not None:
+                self.pool3.destroy(1)
+        except DaosApiError as exc:
+            self.log.info(exc)
+            self.log.info(traceback.format_exc())
+            self.fail("##(2)Pool destroy failed in tearDown.\n")
+        finally:
+            super(PoolSecurityTest, self).tearDown()
+
+    def test_poolconnect(self):
+        """
+        Test basic pool security in pool creation and connect.
+        DAOS-2930: DAOS security: initial tests pool creation with default
+                   ACL and connect.
+        Tests DAOS pools creation, and connect with:
+        (1)pool1: match of owner uid.  (succeed)
+        (2)pool2: match of owner gid.  (succeed)
+        (3)pool3: mismatch of owner uid and gid.
+                                       (permission denied with error -1001)
+
+        :avocado: tags=pool,security_basic,security
+        """
+        user_uid = os.geteuid()
+        user_gid = os.getegid()
+        der_no_permission = "RC: -1001"
+        try:
+            self.log.info(
+                "===Connecting to the DAOS pool1, match of owner uid.")
+            self.log.info("===user_uid:  %s", user_uid)
+            self.log.info("===user_gid:  %s", user_gid)
+            # connect to the pool with permission
+            #   DAOS_PC_RO = int(1 << 0)
+            #   DAOS_PC_RW = int(1 << 1)
+            #   DAOS_PC_EX = int(1 << 2)
+            self.pool1.connect(1 << 1)
```

Comment:
Not Sure if with the new ACLs  DAOS_PC_RO = int(1 << 0), DAOS_PC_EX = int(1 << 2) also prevail. if they do, then why not also test for these?

Reply:
With the new test_utils.py, we use default when connecting the pool.